### DISCUSSION TITLE
Get data format of piped data

### DISCUSSION THREAD

**Discussion by yerlaser** [nushell/nushell#8447] [Q&A] [answered]
I get piped input which can be JSON, YAML or another type that is supported by Nushell and need to process that.
Somehow, Nushell is able to automatically guess the input type because it is showing me a list of tables.
I can then use the corresponding command: `from json`, `from yaml` etc and get that parsed.
I know it detects the format because if I save that data into a file and then `open --raw` it, it shows up as a bunch of strings and even if I pipe that into `lines` I get a different display than if I open it with simple `open`

I would like to automate the format detection.
Is there a way to determine the type of piped input?

**Comment by fdncred** [comment, marked as answer]
I believe format detection is only automated when you `open` a file has an extension that is recognized by nushell like .csv, .yaml, .json, etc. Then nushell automatically calls `from <extension>` where `<extension>` matches your file extension. If you do `open --raw/-r` on a file, the `-r` disables the automatic detection and the automatic `from <extension>` code.

I don't think nushell automatically detects file formats from buffers. For instance, if you have this:
```
> "fname,lname,age\nfirst,last,50\n"
fname,lname,age
first,last,50
```
nushell will just print that to screen even though it looks like a csv. But you could do: 
```
> "fname,lname,age\nfirst,last,50\n" | from csv
╭───┬───────┬───────┬─────╮
│ # │ fname │ lname │ age │
├───┼───────┼───────┼─────┤
│ 0 │ first │ last  │  50 │
╰───┴───────┴───────┴─────╯
```

**Reply by yerlaser** [reply]
Hmm, that makes sense, but somehow seems to contradict my findings. Let me do a bit more research on my data.
Thanks for the answer!
